app/com/linkedin/drelephant/analysis/AnalyticJobGeneratorHadoop2.java

```diff
@@ -59,15 +63,50 @@
       Statistics.MINUTE_IN_MS * 30 + new Random().nextLong() % (3 * Statistics.MINUTE_IN_MS);
 
   private String _resourceManagerAddress;
+  private long _lastRun;
   private long _lastTime = 0;
   private long _fetchStartTime = 0;
   private long _currentTime = 0;
   private long _tokenUpdatedTime = 0;
+  private HadoopSecurity _hadoopSecurity;
   private AuthenticatedURL.Token _token;
   private AuthenticatedURL _authenticatedURL;
   private final ObjectMapper _objectMapper = new ObjectMapper();
 
-  private final Queue<AnalyticJob> _retryQueue = new ConcurrentLinkedQueue<AnalyticJob>();
+  public void fetchAndExecuteJobs(long checkPoint) {
+
+    updateResourceManagerAddresses();
+
+    _lastRun = System.currentTimeMillis();
+    logger.info("Fetching analytic job list...");
+    _hadoopSecurity = ElephantRunner.getInstance().getHadoopSecurity();
+    while (true) {
+      try {
+        _hadoopSecurity.checkLogin();
+        break;
+      } catch (IOException e) {
+        logger.info("Error with hadoop kerberos login", e);
+        getWaitInterval(ElephantRunner.getInstance().getRetryInterval());
+      }
+    }
+
+    _lastTime = checkPoint;
+    List<AnalyticJob> todos;
+    while (true) {
+      try {
+        todos = fetchAnalyticJobs();
+        logger.info("jobs count: " + todos.size());
+        break;
+      } catch (Exception e) {
+        logger.error("Error fetching job list. Try again later...", e);
+        getWaitInterval(ElephantRunner.getInstance().getRetryInterval());
```

Comment:
If it is get, lets not sleep inside the method, or lets rename it to waitFor or something similar.